File: 1567/Solution.go
func getMaxLen(nums []int) int {
    max, n := 0, len(nums)
    for i := 0; i < n; i++ {
        j, neg, leftNeg, rightNeg := i, 0, -1, -1
        for i < n && nums[i] != 0 {
            if nums[i] < 0 {
                neg++
                if leftNeg == -1 {
                    leftNeg = i
                } else {
                    rightNeg = i
                }
            }
            i++
        }
        if neg % 2 == 0 {
            max = maxInt(max, i - j)
        } else{
            leftMax := maxInt(leftNeg, rightNeg) - j
            rightMax := i - leftNeg - 1
            max = maxInt(max, maxInt(leftMax, rightMax))
        }
    }
    return max
}
func maxInt(a, b int) int{
    if a > b {
        return a
    }
    return b
}

File: 1567/Solution.java
class Solution {
    public int getMaxLen(int[] nums) {
        int n = nums.length;
        int max = 0;
        for(int i = 0; i< n; i++){
            int neg = 0, firstNeg = -1, lastNeg = -1, j = i;
            while(i < n && nums[i] != 0){
                if(nums[i] < 0){
                    neg++;
                    if(firstNeg == -1){
                        firstNeg = i;
                    }else{
                        lastNeg = i;
                    }
                } 
                i++;
            }
            if(neg % 2 == 0){
                max = Math.max(max, i - j);
            }else{
                int maxLeft = Math.max(firstNeg, lastNeg) - j;
                int maxRight = i - firstNeg - 1;
                max = Math.max(max, Math.max(maxLeft, maxRight)); 
            }
        }
        return max;

    }
}
